Make nnapi flatten converter accept flex inputs

axitkhurana opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #61025
* **#61024**
* #61023
* #61022
* #61021
* #59529
* #59434

Summary:

As per title

Test Plan:

pytest test/test_nnapi.py::TestNNAPI::test_flatten

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D29480748](https://our.internmc.facebook.com/intern/diff/D29480748)

Commit:
Flatten flex input WIP

Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

axitkhurana commented:
@axitkhurana has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29480748).

Commit:
Update on "Make nnapi flatten converter accept flex inputs"


Summary:

As per title

Test Plan:

pytest test/test_nnapi.py::TestNNAPI::test_flatten

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D29480748](https://our.internmc.facebook.com/intern/diff/D29480748)

[ghstack-poisoned]

axitkhurana commented:
@axitkhurana has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29480748).

Commit:
Update on "Make nnapi flatten converter accept flex inputs"


Summary:

As per title

Test Plan:

pytest test/test_nnapi.py::TestNNAPI::test_flatten

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D29480748](https://our.internmc.facebook.com/intern/diff/D29480748)

[ghstack-poisoned]

dreiss commented on torch/backends/_nnapi/serializer.py:
```diff
@@ -957,23 +957,26 @@ def add_flatten(self, node):
-        #     for dim in out_shape
-        # )
+        if any(dim == 0 for dim in in_oper.shape[start_dim: end_dim + 1]):
+            raise Exception("Flattened dims can't be flexible")
```
I would update this to say "Flattening flexible dims is not supported yet.", because it seems like it should be possible to support this with more work.

Commit:
Update on "Make nnapi flatten converter accept flex inputs"


Summary:

As per title

Test Plan:

pytest test/test_nnapi.py::TestNNAPI::test_flatten

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D29480748](https://our.internmc.facebook.com/intern/diff/D29480748)

[ghstack-poisoned]

Commit:
Update on "Make nnapi flatten converter accept flex inputs"


Summary:

As per title

Test Plan:

pytest test/test_nnapi.py::TestNNAPI::test_flatten

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D29480748](https://our.internmc.facebook.com/intern/diff/D29480748)

[ghstack-poisoned]

axitkhurana commented:
@axitkhurana has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29480748).

Commit:
Update on "Make nnapi flatten converter accept flex inputs"


Summary:

As per title

Test Plan:

pytest test/test_nnapi.py::TestNNAPI::test_flatten

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D29480748](https://our.internmc.facebook.com/intern/diff/D29480748)

[ghstack-poisoned]

axitkhurana commented:
@axitkhurana has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29480748).

Commit:
Update on "Make nnapi flatten converter accept flex inputs"


Summary:

As per title

Test Plan:

pytest test/test_nnapi.py::TestNNAPI::test_flatten

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D29480748](https://our.internmc.facebook.com/intern/diff/D29480748)

[ghstack-poisoned]

axitkhurana commented:
@axitkhurana has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29480748).
